src/gadgetlib2/variable.cpp

```diff
@@ -291,7 +291,12 @@ FElem Variable::eval(const VariableAssignment& assignment) const {
 #ifdef DEBUG
 VariableArray::VariableArray(const string& name) : VariableArrayContents(), name_(name) {}
 VariableArray::VariableArray(const int size, const ::std::string& name) : VariableArrayContents() {
-    for(int i = 0; i < size; ++i) {
+    for (int i = 0; i < size; ++i) {
+        push_back(Variable(GADGETLIB2_FMT("%s[%d]", name.c_str(), i)));
+    }
+}
+VariableArray::VariableArray(const size_t size, const ::std::string& name) : VariableArrayContents() {
```

Comment:
Sounds like it is fine to just change the type;  I was assuming that original signature was also required somewhere but I didn't confirm.